velox/type/Timestamp.h

```diff
@@ -183,19 +185,21 @@ struct Timestamp {
 
   // Keep it in header for getting inlined.
   int64_t toMicros() const {
-    // When an integer overflow occurs in the calculation,
-    // an exception will be thrown.
-    try {
-      return checkedPlus(
-          checkedMultiply(seconds_, (int64_t)1'000'000),
-          (int64_t)(nanos_ / 1'000));
-    } catch (const std::exception& e) {
+    // We use int128_t to make sure the computation does not overflows since
+    // there are cases such that seconds*1000000 does not fit in int64_t,
```

Comment:
Spark is using int64 type to represent micro second, so the max allowed seconds should be `INT64_MAX / 1000000`. For a valid timestamp from Spark, why would `seconds * 1000000` overflow?

Reply:
Would you like to extract this fix to a separate PR like https://github.com/facebookincubator/velox/commit/671e12679ab047d960a99f62d06f78be64134164? We could add test in 'velox/type/tests/TimestampTest.cpp'.

Reply:
sure, let me extract this to a new pr

Reply:
Added here: https://github.com/facebookincubator/velox/pull/11774